src/app/core/local-storage/local-storage.service.ts

```diff
@@ -45,4 +45,32 @@ export class LocalStorageService {
   getItem(key: string) {
     return JSON.parse(localStorage.getItem(`${APP_PREFIX}${key}`));
   }
+
+  /** Returns true if localStorage is defined, otherwise false. */
+  localStorageExists(): boolean {
+    if (localStorage) {
```

Comment:
Just do `return localStorage`

Reply:
@timdeschryver  - When I try that, I get the following typescript error:
`Type 'Storage' is not assignable to type 'boolean'.`

Reply:
@timdeschryver I can do this though, if you'd like
`return (localStorage) ? true : false;`

Reply:
what about `Boolean(localStorage)`?

Reply:
Yes, `Boolean(localStorage)` seems to work.

However, after reading among other things [this SO answer](https://stackoverflow.com/questions/16427636/check-if-localstorage-is-available), I think that checking for localStorage existence is redundant with what the method `testLocalStorage` method does.  The only thing I would change is to remove the test item the method creates.  The method would then look like this:

```typescript
  /** Tests that localStorage exists, can be written to, and read from. */
  testLocalStorage() {
    const testValue = 'testValue';
    const testKey = 'testKey';
    let retrievedValue: string;
    const errorMessage = 'localStorage did not return expected value';

    this.setItem(testKey, testValue);
    retrievedValue = this.getItem(testKey);
    this.removeItem(testKey);

    if (retrievedValue !== testValue) {
      throw new Error(errorMessage);
    }
  }
```
If `localStorage` is unavailable/undefined for any reason, the `set/get/remove` methods will
throw an exception that will be handled by the app error handler (snackbar message displayed and
errors logged in console), which is all we want.

Reply:
OK. I have just pushed in the changes I had outlined above. Let me know what you think.